docs/proposals/0008-ToolAuthAPI.md

```diff
@@ -0,0 +1,459 @@
+# Tool Authorization in Agentic Networking
+
+This proposal defines authorization policies for tool access from AI agents running inside a Kubernetes cluster to MCP servers running in the Kubernetes cluster or outside of the Kubernetes cluster. By default, an AI agent can call initialize, notifications/initialized and tools/list. To enforce a "zero trust" security posture, a tools/call is denied unless it is allowed through the Tool Auth API described in this proposal.
+
+# Non-Goals
+
+The authentication of MCP tool access is not within the scope of this proposal, and will be explored separately in the future.
+
+# Use Cases & Motivation 
```

Comment:
Reference the personas and user journeys described in https://github.com/kubernetes-sigs/kube-agentic-networking/pull/5 after it is merged.